### PR TITLE
Fixes #30997 - allow image_path in all cases

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Fix return value in blank icon edge case.

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
I don't really see how this would be any different. Could you provide a test case to demonstrate?

**Review by lzap — DISMISSED** [review]
I don't mind this, but keep in mind we only have 16 and 64. Letting @ekohl to carry on.

**Comment by ezr-ondrej** [comment]
> I don't really see how this would be any different. Could you provide a test case to demonstrate?

I've added test to describe it better, thanks for pushing me to write it.
In nutshell we've not returned path in this case even if `options = {path: true}`, what caused calls to
`<foreman_url>/">` when UI wanted to fetch the image and that obviously failed :)

**Comment by ezr-ondrej** [comment]
> I don't mind this, but keep in mind we only have 16 and 64. Letting @ekohl to carry on.

My usecase (breadcrumbs testing) is 16x16, but i would argue it is better to fail with valid url in scenarios with invalid size, than with invalid url containing image tag with invalid size :thinking: :smile:

**Review by lzap — DISMISSED** [review]

File: app/helpers/operatingsystems_helper.rb [hunk 1]
```diff
@@ -58,7 +58,10 @@ def icon(record, opts = {})
            when /Windows/i
              "stub/steelblue-w"
            else
-             return os_image_tag(size, "stub/black-x", opts) if record.family.blank?
+             if record.family.blank?
+               return image_path("icons#{size}/stub/black-x.png") if opts[:path]
+               return os_image_tag(size, "stub/black-x", opts)
+             end
              record.family.downcase
```
Comment (ekohl) [inline app/helpers/operatingsystems_helper.rb]: How about
```ruby
if record.family.blank?
  'stub/black-x'
else
  record.family.downcase
end
```

I think that's essentially the same, but keeps the structure of the method more more the same.

**Review by tbrisker — CHANGES_REQUESTED** [review]
See inline request from @ekohl

File: app/helpers/operatingsystems_helper.rb (hunk 1 above)
Comment (ezr-ondrej) [reply app/helpers/operatingsystems_helper.rb]: Nice improvement! :+1:

**Review by tbrisker — APPROVED** [review]

**Review by ekohl — APPROVED** [review]

**Comment by tbrisker** [comment]
Thanks @ezr-ondrej !

**Pull request merged**